### PR TITLE
Fileinput

### PR DESCRIPTION
**Pull request by RoronoaSrinu**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
                       jcr:primaryType="nt:unstructured"
                       jcr:title="formcontainer-policy"
                       sling:resourceType="wcm/core/components/policy/policy"
-                      components="[/apps/forms-components-examples/components/form/textinput,/apps/forms-components-examples/components/form/datepicker,/apps/forms-components-examples/components/form/panelcontainer]">
+                      components="[/apps/forms-components-examples/components/form/textinput,/apps/forms-components-examples/components/form/datepicker,/apps/forms-components-examples/components/form/panelcontainer, /libs/core/fd/components/form/fileinput/v1/fileinput]">
```
Comment (gahuja1991) [inline it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml:15]: We would need to add the /apps path here.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/site/css/fileinputview.css [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
+## BEM Description
+BLOCK cmp-adaptiveform-fileinput
+    ELEMENT cmp-adaptiveform-textinput__label
```
Comment (gahuja1991) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/site/css/fileinputview.css:4]: Please change these name as well

Comment (RoronoaSrinu) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/site/css/fileinputview.css]: Done

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/editor/js/editDialog.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        hideAndShowElements();
+        component.on("change", function() {
+            hideAndShowElements();
+        });
+
+        if(component.checked){
+         fileinputType.attributes.value.value="file[]";
+         }
+        else{
+         fileinputType.attributes.value.value="file";
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/clientlibs/editor/js/editDialog.js:53]: minor: indentation

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/_cq_dialog/.content.xml [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+                                    fieldDescription="Error message shown when more than maximum File Size files are uploaded"
+                                    granite:class="cmp-adaptiveform-fileinput__maxfilesizemessage"
+                                    fieldLabel="Max File Size Error Message"
+                                    name="./maxFileSizeMessage"/>
+                            <acceptMessage
+                                    jcr:primaryType="nt:unstructured"
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+                                    fieldDescription="Error message shown when different file type are uploaded which are not included in Supported File Types"
+                                    granite:class="cmp-adaptiveform-fileinput__acceptemessage"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/_cq_dialog/.content.xml:155]: typo: `acceptmessage`

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    ELEMENT cmp-adaptiveform-fileinput__errormessage
+```
+
+## JavaScript Data Attribute Bindings
+
+Apply a `data-cmp-is="adaptiveFormTextInput"` attribute to the wrapper block to enable initialization of the JavaScript component.
+
+The following attributes can be added to the same element to provide options:
+
+1. `data-cmp-required-message` - populated with requiredMessage from the component's edit dialog
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md:67]: pls remove required message attribute.
adaptive form container path attribute missing in readme.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+The component provides a `core.forms.components.fileinput.v1` client library category that contains a JavaScript
+component. It should be added to a relevant site client library using the `embed` property.
+
+It also provides a `core.forms.components.fileinput.v1.editor` editor client library category that includes
+JavaScript handling for dialog interaction. It is already included by its edit dialog.
+
+## BEM Description
+```
+BLOCK cmp-adaptiveform-textinput
+    ELEMENT cmp-adaptiveform-fileinput__label
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md:53]: validation messages elements classes missing from doc.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+The following properties are written to JCR for this Form File component and are expected to be available as `Resource` properties:
+
+1. `./jcr:title` - defines the label to use for this field
+2. `./hideTitle` - if set to `true`, the label of this field will be hidden
+3. `./name` - defines the name of the field, which will be submitted with the form data
+4. `./default` - defines the default value of the field
+5. `./description` - defines a help message that can be rendered in the field as a hint for the user
+6. `./required` - if set to `true`, this field will be marked as required, not allowing the form to be submitted until the field has a value
+7. `./requiredMessage` - defines the message displayed as tooltip when submitting the form if the value is left empty
+8. `./readOnly` - if set to `true`, the filed will be read only
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md:41]: some more properties are also there in dialog. pls capture those as well in readme.

File: it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml [hunk 8]
```diff
@@ -12,7 +12,7 @@
                       jcr:primaryType="nt:unstructured"
                       jcr:title="formcontainer-policy"
                       sling:resourceType="wcm/core/components/policy/policy"
-                      components="[/apps/forms-components-examples/components/form/textinput,/apps/forms-components-examples/components/form/datepicker,/apps/forms-components-examples/components/form/panelcontainer]">
+                      components="[/apps/forms-components-examples/components/form/textinput,/apps/forms-components-examples/components/form/datepicker,/apps/forms-components-examples/components/form/panelcontainer,/apps/forms-components-examples/components/form/fileinput,/libs/core/fd/components/form/fileinput/v1/fileinput,/libs/core/fd/components/form/textinput/v1/textinput]">
```
Comment (nit23uec) [inline it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml:15]: it should be /apps path

File: bundles/core/src/test/resources/aemform/test-content.json [hunk 9]
```diff
@@ -6,7 +6,7 @@
             "jcr:title"         : "AEM Form Test",
             "sling:resourceType": "core/wcm/components/page/v2/page",
             "jcr:description"   : "AEM Form description",
-            "sling:configRef"   : "/conf/core-components-examples/content",
+            "sling:configRef"   : "/conf/fileinput/content",
```
Comment (nit23uec) [inline bundles/core/src/test/resources/aemform/test-content.json:9]: is this change really required?
cc @rismehta

**Review by nit23uec — CHANGES_REQUESTED** [review]
inline comments.

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/form/Base.java [hunk 10]
```diff
@@ -128,6 +128,8 @@ public enum ConstraintType {
         MAX_ITEMS("maxItems"),
         UNIQUE_ITEMS("uniqueItems"),
         ENFORCE_ENUM("enforceEnum"),
+        MAXFILESIZE("maxFileSize"),
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/form/Base.java:131]: Based on camel case, add underscore

File: bundles/core/src/test/resources/aemform/test-content.json (hunk 9 above)
Comment (rismehta) [reply bundles/core/src/test/resources/aemform/test-content.json:9]: no

**Review by rismehta — COMMENTED** [review]
-1 for now

File: it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml (hunk 8 above)
Comment (RoronoaSrinu) [reply it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml:15]: do you mean remove this /libs/core/fd/components/form/fileinput/v1/fileinput and add only /apps/forms-components-examples/components/form/fileinput

Comment (nit23uec) [reply it/content/src/main/content/jcr_root/conf/core-components-it/settings/wcm/policies/.content.xml:15]: yes.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+```
+BLOCK cmp-adaptiveform-fileinput
+    ELEMENT cmp-adaptiveform-fileinput__label
+    ELEMENT cmp-adaptiveform-fileinput__widget
+    ELEMENT cmp-adaptiveform-fileinput__questionmark
+    ELEMENT cmp-adaptiveform-fileinput__shortdescription
+    ELEMENT cmp-adaptiveform-fileinput__longdescription
+    ELEMENT cmp-adaptiveform-fileinput__errormessage
+    ELEMENT cmp-adaptiveform-fileinput__minmessage
+    ELEMENT cmp-adaptiveform-fileinput__maxmessage
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md:69]: we don't need to add the edit dialog classes here. only those classes should be added those can be styled in the actual rendition.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/fileinput.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/fileinput.js:2]: copyright headers. 2022.

File: ui.tests/test-module/specs/fileinput.spec.js [hunk 13]
```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright 2022 Adobe Systems Incorporated
```
Comment (nit23uec) [inline ui.tests/test-module/specs/fileinput.spec.js:2]: new copyright headers should be used.

**Review by nit23uec — CHANGES_REQUESTED** [review]
inline comments.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+## JavaScript Data Attribute Bindings
+
+Apply a `data-cmp-is="adaptiveFormFileInput"` attribute to the wrapper block to enable initialization of the JavaScript component.
+
+The following attributes can be added to the same element to provide options:
+
+A `data-cmp-hook-adaptiveform-fileinput="input"` attribute should be added to the input  so that the JavaScript is able to target it.
+
+Apply a `data-cmp-is="adaptiveFormFileInput"` and `data-cmp-adaptiveformcontainer-path="/content/forms/af/fileinput-test/jcr:content/guideContainer"` attribute to the wrapper block to enable initialization of the JavaScript component.
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md:81]: data-cmp-adaptiveformcontainer-path="{formStructParser.xxx}"

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+    ELEMENT cmp-adaptiveform-fileinput__acceptmessage
+```
+
+## JavaScript Data Attribute Bindings
+
+Apply a `data-cmp-is="adaptiveFormFileInput"` attribute to the wrapper block to enable initialization of the JavaScript component.
+
+The following attributes can be added to the same element to provide options:
+
+A `data-cmp-hook-adaptiveform-fileinput="input"` attribute should be added to the input  so that the JavaScript is able to target it.
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/README.md:79]: pls remove.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/_cq_dialog/.content.xml [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+                                    fieldDescription="Error message shown when less than minimum Files are uploaded"
+                                    granite:class="cmp-adaptiveform-fileinput__minmessage"
+                                    fieldLabel="Min Files Error Message"
+                                    name="./minItemsMessage"/>
+                            <maxFilesMessage
+                                    jcr:primaryType="nt:unstructured"
+                                    sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
+                                    fieldDescription="Error message shown when more than maximum Files are uploaded"
+                                    granite:class="cmp-adaptiveform-fileinput__maxmessage"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/fileinput/v1/fileinput/_cq_dialog/.content.xml:141]: pls remove granite:class is element not used in editDialog.js

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/form/Base.java [hunk 17]
```diff
@@ -18,23 +18,220 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonValue;
 
 /**
  * A base interface to be extended by all the different types of form elements.
  *
  * @since com.adobe.cq.forms.core.components.models.form 0.0.1
  */
 @ConsumerType
-public interface Base extends FormComponent {
```
Comment (nit23uec) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/models/form/Base.java:35]: this reverts our changes. pls rebase from dev, and accept the dev changes.

**Review by nit23uec — APPROVED** [review]
